style/topic.css

```diff
@@ -10,39 +10,67 @@
  * governing permissions and limitations under the License.
  */
 
+ .topic-page {
+  /* header bg colors */
+  --bg1: #2c0167;
+  --bg2: #fd8467;
+  --bg3: #83106a;
+}
+
 .topic-page header,
 .topic-page main,
 .topic-page footer {
   visibility: visible;
 }
 
-.topic-page main .default:not(.articles) {
-  padding: 0;
-}
-
 .topic-page  .articles {
   background-color: white;
 }
 
+.topic-page .articles-empty {
+  visibility: hidden;
+}
+
 .topic-page .articles-empty::before {
   content: 'No articles found in this category';
 }
 
-.topic-page {
-  /* header bg colors */
-  --bg1: #2c0167;
-  --bg2: #fd8467;
-  --bg3: #83106a;
+.topic-page .articles-empty.user-filters {
+  visibility: visible;
+}
+
+.topic-page .articles-empty.user-filters::before {
+  content: "Sorry, no matching results";
```

Comment:
What about localisation ?

Reply:
Good catch! This part of the CSS would actually not be used until #431 will be merged. I removed the code from this PR again.